test/test_interpreter.cpp

```diff
@@ -626,6 +662,49 @@ TEST_SUITE("xmagics_apply"){
         manager.apply("%dummy", kernel_res);
         REQUIRE(kernel_res["status"] == "error");
     } 
+
+    TEST_CASE("good_status_line") {
+
+        xcpp::xpreamble_manager preamble_manager;
+
+        preamble_manager.register_preamble("magics", new xcpp::xmagics_manager());
+
+        preamble_manager["magics"].get_cast<xcpp::xmagics_manager>().register_magic("magic2", MyMagicCell());
+
+        nl::json kernel_res;
+
+        preamble_manager["magics"].get_cast<xcpp::xmagics_manager>().apply("%%magic2 qwerty", kernel_res);
+
+        REQUIRE(kernel_res["status"] == "ok");
+    } 
+
+    TEST_CASE("good_status_cell") {
+
+        xcpp::xpreamble_manager preamble_manager;
+
+        preamble_manager.register_preamble("magics", new xcpp::xmagics_manager());
+
+        preamble_manager["magics"].get_cast<xcpp::xmagics_manager>().register_magic("magic1", MyMagicLine());
+
+        nl::json kernel_res;
+
+        preamble_manager["magics"].get_cast<xcpp::xmagics_manager>().apply("%magic1 qwerty", kernel_res);
+
+        REQUIRE(kernel_res["status"] == "ok");
+    } 
+
+    TEST_CASE("cell magic with empty cell body") {
+        xcpp::xmagics_manager manager;
+
+        std::stringstream ss;
```

Comment:
The stringstream should go in the StdErrRedirect object. I'd also like to rename it as `StreamRedirectRAII` which takes a parameter which stream to be redirected.